src/main/java/javaslang/control/Option.java

```diff
@@ -53,6 +53,19 @@
     }
 
     /**
+     *
+     * Creates {@code Some} of suppliers value if condition is true, or {@code None} in other case
+     *
+     * @return return {@code Some} of supplier's value if condition is true, or {@code None} in other case
+     */
+    static <T> Option<T> when(boolean condition, Supplier<T> supplier) {
+        if (!condition) {
+            return none();
+        }
+        return of(supplier.get());
```

Comment:
I would to a `condition ? of(supplier.get()) : none();` here but that is a matter of taste. Using your guard method is ok!